File: backend/src/main/java/com/pickleball/backend/modules/booking/service/BookingService.java
package com.pickleball.backend.modules.booking.service;

import com.pickleball.backend.modules.booking.dto.request.CreateBookingRequest;
import com.pickleball.backend.modules.booking.dto.response.BookingResponse;
import com.pickleball.backend.modules.booking.dto.response.CourtScheduleResponse;
import com.pickleball.backend.response.PageResponse;

import java.time.LocalDate;

public interface BookingService {

    BookingResponse createBooking(String userEmail, CreateBookingRequest request);

    java.util.List<BookingResponse> createBulkBookings(String userEmail, com.pickleball.backend.modules.booking.dto.request.BulkCreateBookingRequest request);
    
    void payBooking(String userEmail, Long bookingId);

    void cancelBooking(String userEmail, Long bookingId);

    PageResponse<BookingResponse> getMyBookings(String userEmail, int page, Integer size);

    PageResponse<BookingResponse> getClubBookings(String userEmail, int page, Integer size);

    CourtScheduleResponse getCourtSchedule(Long courtId, LocalDate bookingDate);
}

File: frontend/src/services/booking.service.ts
import { apiClient } from "@/lib/axios";
import { PageResponse } from "@/types/common";
import { Booking, CourtSchedule, CreateBookingRequest } from "@/types/booking.type";

export const bookingService = {
  createBooking: async (request: CreateBookingRequest) => {
    const res = await apiClient.post<{ data: Booking }>("/bookings", request) as unknown as { data: Booking };
    return res.data;
  },

  getMyBookings: async (params?: { page?: number; size?: number }) => {
    const res = await apiClient.get<{ data: PageResponse<Booking> }>("/bookings/my-bookings", {
      params,
    }) as unknown as { data: PageResponse<Booking> };
    return res.data;
  },

  getClubBookings: async (params?: { page?: number; size?: number }) => {
    const res = await apiClient.get<{ data: PageResponse<Booking> }>("/bookings/club", {
      params,
    }) as unknown as { data: PageResponse<Booking> };
    return res.data;
  },

  cancelBooking: async (id: number) => {
    const res = await apiClient.delete<{ data: null }>(`/bookings/${id}`) as unknown as { data: null };
    return res.data;
  },

  payBooking: async (id: number) => {
    const res = await apiClient.post<{ data: null }>(`/bookings/${id}/pay`) as unknown as { data: null };
    return res.data;
  },

  getCourtSchedule: async (courtId: number, date: string) => {
    const res = await apiClient.get<{ data: CourtSchedule }>(`/bookings/court/${courtId}`, {
      params: { date },
    }) as unknown as { data: CourtSchedule };
    return res.data;
  },
};
